components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -299,6 +314,14 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
     this.conflictError = false;
   }
 
+  handleConditionChange(): void {
+    if (this.editingRule) {
+      // changes on the condition forms do not bubble up to the ruleForm
+      // so we explicitly set it to dirty here
```

Comment:
Aren't nested formgroups a thing? I would, too, prefer if we didn't have to manage this manually...

Reply:
yeah I would have expected it to work too. Maybe worth digging into the formgroup docs